self_enrollment_tool/views.py

```diff
@@ -0,0 +1,177 @@
+from django.shortcuts import render
+
+import logging
+
+from canvas_sdk.exceptions import CanvasAPIError
+from canvas_sdk.methods import courses
+from django.conf import settings
+from django.contrib import messages
+from django.contrib.auth.decorators import login_required
+from django.shortcuts import render
+from django.views.decorators.http import require_http_methods
+from django_auth_lti import const
+from django_auth_lti.decorators import lti_role_required
+from icommons_common.canvas_utils import SessionInactivityExpirationRC
+from icommons_common.models import CourseInstance
+from lti_permissions.decorators import lti_permission_required
+from self_enrollment_tool.models import SelfEnrollmentCourse
+
+
+logger = logging.getLogger(__name__)
+
+SDK_CONTEXT = SessionInactivityExpirationRC(**settings.CANVAS_SDK_SETTINGS)
+
+@login_required
+@lti_role_required(const.ADMINISTRATOR)
+@lti_permission_required(settings.PERMISSION_SELF_ENROLLMENT_TOOL)
+@require_http_methods(['GET'])
+def index(request):
+    context = {}
+    return render(request, 'self_enrollment_tool/index.html', context)
+
+
+@login_required
+@lti_role_required(const.ADMINISTRATOR)
+@lti_permission_required(settings.PERMISSION_CANVAS_SITE_DELETION)
+@require_http_methods(['GET', 'POST'])
+def lookup(request):
+    course_search_term = request.POST.get('course_search_term')
+    course_search_term = course_search_term.strip()
+    context = {
+        'canvas_url': settings.CANVAS_URL,
+        'abort': False,
+    }
+
+    if course_search_term.isnumeric():
```

Comment:
There are a handful of logging statements at the error or exception level. This should probably be turned down to a warning level for some